src/librustc_mir/monomorphize/polymorphize.rs

```diff
@@ -285,10 +285,11 @@ impl<'a, 'tcx> TypeVisitor<'tcx> for UsedGenericParametersVisitor<'a, 'tcx> {
                     "visit_ty: used_parameters={:064b} results={:064b}",
                     self.used_parameters, results
                 );
-                // Can just `|` the results of running the analysis on the closure/generator, the
-                // synthetic parameters will be marked as used and be the rightmost bits, so
-                // won't make a difference.
-                *self.used_parameters |= !results;
+                for (i, arg) in substs.iter().enumerate() {
+                    if !is_bit_set(results, i.try_into().unwrap()) {
```

Comment:
`results` could be named `unused` (or something even more descriptive), as it stands it's not clear what this condition does.

Reply:
Fixed!